controllers/humiocluster_controller.go

```diff
@@ -1515,17 +1515,32 @@ func (r *HumioClusterReconciler) ensureLicenseIsValid(ctx context.Context, hc *h
 func (r *HumioClusterReconciler) ensureLicense(ctx context.Context, hc *humiov1alpha1.HumioCluster, req ctrl.Request) (reconcile.Result, error) {
 	r.Log.Info("ensuring license")
 
-	cluster, err := helpers.NewCluster(ctx, r, hc.Name, "", hc.Namespace, helpers.UseCertManager())
-	if err != nil {
-		return r.ensureInitialLicense(ctx, hc, r.HumioClient.GetBaseURL(hc), req)
+	humioAPIConfig := &humioapi.Config{
+		Address: r.HumioClient.GetBaseURL(hc),
+	}
+
+	// Get CA
+	if helpers.TLSEnabled(hc) {
+		existingCABundle, err := kubernetes.GetSecret(ctx, r, constructClusterCACertificateBundle(hc).Spec.SecretName, hc.Namespace)
+		if errors.IsNotFound(err) {
+			r.Log.Info("waiting for secret with CA bundle")
+			return reconcile.Result{Requeue: true, RequeueAfter: time.Second * 10}, nil
+		}
+		if err != nil {
+			r.Log.Error(err, "unable to obtain CA certificate")
+			return reconcile.Result{Requeue: true, RequeueAfter: time.Second * 10}, err
+		}
+		humioAPIConfig.CACertificatePEM = string(existingCABundle.Data["ca.crt"])
 	}
-	r.HumioClient.SetHumioClientConfig(cluster.Config(), req)
+	r.HumioClient.SetHumioClientConfig(humioAPIConfig, req)
```

Comment:
I think we should configure the HumioClient in the topmost part of the Reconcile method, right after we fetch the HC configuration.

Reply:
I don't think we can do that since we must wait for the certs, but we can probably refactor this into a method.